### PR TITLE
Reduce the config parsing codes (-360 Lines) for DS2 and make it looks cleaner.

### LINKED ISSUE #226: Reduce the config parsing codes for DS2 and make it looks cleaner.
Shorten the code:
```
parser.add_argument(
    "--num_passes",
    default=200,
    type=int,
    help="Training pass number. (default: %(default)s)")
```
to
```
add_arg('num_passes',       int,    200,    "# of training epochs.")
```

And align the columns to make it looks cleaner.

### PR DESCRIPTION
**Pull request by xinghai-sun**
Resolved #226

### CODE REVIEW

**Comment by xinghai-sun** [comment]
Should we allow more than 80 columns for this part of codes, making it look better?

It will look like:
screen shot 2017-09-05 at 12 44 47 am

Another choice is that we just forget about the column alignment and follows the normal python coding style.

File: deep_speech_2/demo_server.py [hunk 1]
```diff
@@ -9,118 +9,74 @@
...
+        help=help + ' Default: %(default)s.',
+        **kwargs)
+
+
+# yapf: disable
+# configurations of overall
+add_arg('host_port',        int,    8086,    "Server's IP port.")
+add_arg('host_ip',          str,
+        'localhost',
+        "Server's IP address.")
```
Comment (xinghai-sun) [inline deep_speech_2/demo_server.py]: Should we allow more than 80 columns of codes in this part to make it look better?

File: deep_speech_2/demo_server.py [hunk 2]
```diff
@@ -9,118 +9,67 @@
 import struct
 import wave
 import paddle.v2 as paddle
-from utils import print_arguments
 from data_utils.data import DataGenerator
 from model import DeepSpeech2Model
 from data_utils.utils import read_manifest
 
+
+def add_arg(argname, type, default, help, **kwargs):
```
Comment (kuke) [inline deep_speech_2/demo_server.py]: Can we move down or out this function?

File: deep_speech_2/evaluate.py [hunk 3]
```diff
@@ -5,145 +5,81 @@
...
-    default='wer',
-    choices=['wer', 'cer'],
-    type=str,
-    help="Error rate type for evaluation. 'wer' for word error rate and 'cer' "
-    "for character error rate. "
-    "(default: %(default)s)")
+add_arg('batch_size',       int,    128,    "Minibatch size.")
+add_arg('trainer_count',    int,    8,      "# of Trainers (CPUs or GPUs).")
+add_arg('beam_size',        int,    500,    "Beam search width.")
+add_arg('parallels_bsearch',int,    12,     "# of CPUs for beam search.")
```
Comment (kuke) [inline deep_speech_2/evaluate.py]: Maybe we'd better keep the prefix ```num_``` here, can use ```num_proc_beam_search``` or something like that.

File: deep_speech_2/evaluate.py [hunk 4]
```diff
@@ -5,145 +5,81 @@
...
+        './checkpoints/params.latest.tar.gz',
+        "If None, the training starts from scratch, "
+        "otherwise, it resumes from the pre-trained model.")
+add_arg('lang_model_path',  str,
+        'lm/data/common_crawl_00.prune01111.trie.klm',
+        "Filepath for language model.")
+add_arg('decoder_method',   str,
+        'ctc_beam_search',
+        "Decoder method. Options: ctc_beam_search, ctc_greedy",
+        choices = ['ctc_beam_search', 'ctc_greedy'])
```
Comment (kuke) [inline deep_speech_2/evaluate.py]: Should ```beam_search``` and ```greedy```  be enough?

File: deep_speech_2/demo_server.py [hunk 5]
```diff
@@ -9,118 +9,67 @@
...
-    "in beam search. (default: %(default)f)")
+add_arg('host_port',        int,    8086,    "Server's IP port.")
+add_arg('beam_size',        int,    500,    "Beam search width.")
+add_arg('num_conv_layers',  int,    2,      "# of convolution layers.")
+add_arg('num_rnn_layers',   int,    3,      "# of recurrent layers.")
+add_arg('rnn_layer_size',   int,    2048,   "# of recurrent cells per layer.")
+add_arg('alpha',            float,  0.36,   "Coef of LM for beam search.")
+add_arg('beta',             float,  0.25,   "Coef of WC for beam search.")
+add_arg('cutoff_prob',      float,  0.99,   "Cutoff probability for pruning.")
+add_arg('use_gru',          bool,   False,  "Use GRUs instead of Simple RNNs.")
```
Comment (kuke) [inline deep_speech_2/demo_server.py]: ```Simple``` -> ```simple```

File: deep_speech_2/evaluate.py [hunk 6]
```diff
@@ -5,145 +5,81 @@
...
+        'datasets/vocab/eng_vocab.txt',
+        "Filepath of vocabulary.")
+add_arg('model_path',       str,
+        './checkpoints/params.latest.tar.gz',
+        "If None, the training starts from scratch, "
+        "otherwise, it resumes from the pre-trained model.")
+add_arg('lang_model_path',  str,
+        'lm/data/common_crawl_00.prune01111.trie.klm',
+        "Filepath for language model.")
+add_arg('decoder_method',   str,
```
Comment (kuke) [inline deep_speech_2/evaluate.py]: Maybe ```decode_method``` or ```decoding_method``` would be better.

File: deep_speech_2/evaluate.py [hunk 7]
```diff
@@ -3,147 +3,74 @@
...
-    type=str,
-    help="Error rate type for evaluation. 'wer' for word error rate and 'cer' "
-    "for character error rate. "
-    "(default: %(default)s)")
+add_arg = functools.partial(add_arguments, argparser=parser)
+# yapf: disable
+add_arg('batch_size',       int,    128,    "Minibatch size.")
+add_arg('trainer_count',    int,    8,      "# of Trainers (CPUs or GPUs).")
+add_arg('beam_size',        int,    500,    "Beam search width.")
+add_arg('parallels_bsearch',int,    12,     "# of CPUs for beam search.")
```
Comment (kuke) [inline deep_speech_2/evaluate.py]: Maybe we'd better keep the prefix ```num_```, and can use ```num_proc_bsearch```, ```num_proc_data``` or something like that.

**Review by kuke — COMMENTED** [review]
Almost LGTM

**Review by pkuyym — APPROVED** [review]
LGTM++

File: deep_speech_2/demo_server.py (hunk 5 above)
Comment (xinghai-sun) [reply deep_speech_2/demo_server.py]: Done.

File: deep_speech_2/evaluate.py (hunk 7 above)
Comment (xinghai-sun) [reply deep_speech_2/evaluate.py]: Done.

File: deep_speech_2/evaluate.py (hunk 6 above)
Comment (xinghai-sun) [reply deep_speech_2/evaluate.py]: Done.

File: deep_speech_2/evaluate.py (hunk 4 above)
Comment (xinghai-sun) [reply deep_speech_2/evaluate.py]: Add "ctc" to be more specific.

**Pull request merged**